src/Utility.cc

```diff
@@ -120,6 +120,21 @@ namespace ignition
         pose);
     }
 
+    /////////////////////////////////////////////
+    math::SphericalCoordinates Convert(const msgs::SphericalCoordinates &_sc)
+    {
+      math::SphericalCoordinates out;
+      if (_sc.surface_model() == msgs::SphericalCoordinates::EARTH_WGS84)
+      {
+        out.SetSurface(math::SphericalCoordinates::EARTH_WGS84);
+      }
+      out.SetLatitudeReference(IGN_DTOR(_sc.latitude_deg()));
```

Comment:
Does it make sense to continue doing the conversion if the `surface_model` isn't supported? I think we shouldn't do it as the values will be probably incorrect.

Reply:
The `ign-math` class is populated with default values, so the caller wouldn't know if we're populating the message or leaving the default values. So instead of skipping the rest of the message, I'm printing a warning on c11ebb3480b4708aa5932e87c53f9a698f95895b. 

For the messages, the field is left empty, and I also printed a warning, but still populated the rest of the function.

I think that ultimately, it shouldn't be the conversion function's responsibility to decide what are valid combinations. It should convert as much as possible and the caller should be the one deciding what is acceptable. What do you think?

Reply:
Practically speaking I don't see any problems right now because we only have one surface type in Ignition Msgs, so it's impossible to receive an unsupported type. I agree that potentially receiving a default `math::SphericalCoordinates` is ambiguous but returning a `math::SphericalCoordinates` with default surface type (`WGS_84`) but potentially lat/lon, heading and elevation references coming from a different surface type could be hard to debug. The error message definitely helps.

I'm approving but as an idea, we could create a new type in `ignition::math::SurfaceType` that is `Unsupported` or similar. We could use this type in this conversion for example, when we detect an unsupported type set in `ignition::msgs::SphericalCoordinates`. This way, after calling `convert()` we can programmatically check if the conversion was right.

Reply:
Yup, makes sense. Ticketed https://github.com/ignitionrobotics/ign-math/issues/237.